File: backend/src/GarminFilter.Api/Models/AppQueryModel.cs
using GarminFilter.SharedKernel.App.ValueObjects;

namespace GarminFilter.Api.Models;

public class AppQueryModel
{
	public required HashSet<AppPermission> ExcludePermissions { get; set; }
	public bool? Paid { get; set; }
	public string? Developer { get; set; }
	public int PageIndex { get; set; }
	public int PageSize { get; set; }
	public AppOrder OrderBy { get; set; } = AppOrders.ReviewCount;
}

File: frontend/lib/models/app_query_model.dart
class AppQueryModel {
  final List<String> excludePermissions;
  final bool? paid;
  final String? developer;
  final int pageIndex;
  final int pageSize;
  final String orderBy;

  AppQueryModel({required this.excludePermissions, required this.paid, required this.developer, required this.pageIndex, required this.pageSize, required this.orderBy});

  Map<String, dynamic> toJson() {
    return {'excludePermissions': excludePermissions, 'paid': paid, 'developer': developer, 'pageIndex': pageIndex, 'pageSize': pageSize, 'orderBy': orderBy};
  }
}
